pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -30,12 +30,23 @@ func updateAIStatusUUID(ctx *zedmanagerContext, uuidStr string) {
 		removeAIStatus(ctx, status)
 		return
 	}
-	changed := doUpdate(ctx, *config, status)
-	if changed {
-		log.Functionf("updateAIStatusUUID status change %d for %s",
-			status.State, uuidStr)
-		publishAppInstanceStatus(ctx, status)
-		publishAppInstanceSummary(ctx)
+
+	if config.IsDesignatedNodeID {
+		changed := doUpdate(ctx, *config, status)
+		if changed {
+			log.Functionf("updateAIStatusUUID status change %d for %s",
+				status.State, uuidStr)
+			publishAppInstanceStatus(ctx, status)
+			publishAppInstanceSummary(ctx)
+		}
+	} else {
+		// doActivate
+		changed := doActivate(ctx, uuidStr, *config, status)
```

Comment:
Isn't the handling the effectiveactivate inside the 'doUpdate', we don't see a problem for non-DNid device?

Reply:
For non-DNId device the volumes and content should be ready by now. Only activate the app.  Typically this is called from your cluster pubsub now. We might have to rework a bit after Yuri puts the app instance code into zedcloud.

Reply:
I did not any reason to rework this code. This is working with Yuris API changes.